### PR TITLE
Fix issue in EliminateConcatStridedSlice with compilation problem

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
Fix errors 
 - 1) the concat & StridedSlice all can be eliminated;
 - 2) the inputs of concat have dim value = 1 in concat axis
 - 2) Fix the bug caused by EliminateConcatStridedSlice Pass which did not consider the Nodes which do not have default output index, such as TopK.

### Tickets:
 - CVS-165105
 - CVS-164371
 - CVS-165403

### CODE REVIEW

File: src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp [hunk 1]
```diff
@@ -584,6 +584,8 @@ pass::EliminateConcatStridedSlice::EliminateConcatStridedSlice() {
                 if (end_constant_node == nullptr)
                     return false;
                 auto end_values = end_constant_node->cast_vector<int64_t>();
+                if (end_values[concat_axis] == INT64_MAX)
```
Comment (praasz) [inline src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp:587]:
```suggestion
                if (end_values[concat_axis] == std::numeric_limits<int64_t>::max())
```

Add tests, which reproduce reported issue, please.

**Comment by dorloff** [comment]
@zhaixuejun1993 , could you please make the commit message and description more informative? The commit message should reflect what exactly was done, not the ticket number.

File: src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp [hunk 2]
```diff
@@ -584,6 +584,8 @@ pass::EliminateConcatStridedSlice::EliminateConcatStridedSlice() {
                 if (end_constant_node == nullptr)
                     return false;
                 auto end_values = end_constant_node->cast_vector<int64_t>();
+                if (end_values[concat_axis] == std::numeric_limits<int64_t>::max())
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp:587]: it believe it's better to check that `end_values[concat_axis] > concat->get_shape()[concat_axis]` , `end_values[concat_axis] == std::numeric_limits<int64_t>::max()` is a corner case

according to the strided slice specification
{5E2F3BD7-A6F2-4A53-AEB4-7EE5D4282D72}

File: src/common/transformations/tests/common_optimizations/nop_elimination.cpp [hunk 3]
```diff
@@ -1598,6 +1598,51 @@ TEST_F(TransformationTestsF, EliminateConcatStridedSlice) {
     }
 }
 
+TEST_F(TransformationTestsF, EliminateConcatStridedSliceAll) {
```
Comment (itikhono) [inline src/common/transformations/tests/common_optimizations/nop_elimination.cpp:1601]: as I can see, several issues were fixed in this PR, but only one test was implemented
does this test cover all fixed cases? if not, could you please extend test coverage?

File: src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp [hunk 4]
```diff
@@ -653,6 +657,8 @@ pass::EliminateConcatStridedSlice::EliminateConcatStridedSlice() {
                 new_concat_in_nodes.push_back(concat_input_node);
             }
         }
+        if (new_concat_in_nodes.size() == concat_inputs.size())
+            return false;
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp:661]: are you sure about `return false` here?
it looks like, the graph may have changed earlier
{0FF8C86F-121E-49DB-8D64-F708CDFA1027}
we usually `return true` if the graph structure has been changed.

File: src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp (hunk 2 above)
Comment (zhaixuejun1993) [reply src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp]: Good idea

File: src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp [hunk 5]
```diff
@@ -619,18 +621,23 @@ pass::EliminateConcatStridedSlice::EliminateConcatStridedSlice() {
             if (!matched)
                 mismatch_slices.push_back(std::make_tuple(slice_node, slice_begin, slice_end));
         }
+        if (mismatch_slices.empty())
+            return true;
+
+        if (mismatch_slices.size() == slice_out_index_in_concat.size())
+            return false;
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp:628]: I looks like we need to track if this `replace_output_update_name` call changed the graph or not
{21E1FFBE-0E1E-427D-930B-E58AFE8419B3}
and return true/false depending on that here
{542A918E-AA47-4161-B2D4-E9CB677DDE4B}

**Review by itikhono — APPROVED** [review]

**Review by praasz — APPROVED** [review]
